src/LibraryManager.Vsix/UI/Controls/Library.xaml.cs

```diff
@@ -284,5 +291,39 @@ private void Library_LostFocus(object sender, RoutedEventArgs e)
                 Flyout.IsOpen = false;
             }
         }
+
+        private string GetLibraryName(string insertionText)
+        {
+            if (string.IsNullOrEmpty(insertionText))
+            {
+                return string.Empty;
+            }
+
+            int atIndex = insertionText.IndexOf('@', 1);
+
+            if (atIndex > -1)
+            {
+                return insertionText.Substring(0, atIndex);
+            }
+            else
+            {
+                return insertionText;
+            }
+        }
+
+        private bool IsValidLibrary(IEnumerable<CompletionItem> items)
+        {
+            if (items.Count() == 1)
+            {
+                CompletionItem item = items.First();
+
+                if (item.InsertionText.Equals(Text))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
```

Comment:
Why are we duplicating the code here? You can get the name from SelectedPackage in ViewModel. Let's sync up offline.

Reply:
As discussed, using Installdialogviewmodel instead

---
In reply to: [200697404](https://github.com/aspnet/LibraryManager/pull/285#discussion_r200697404) [](ancestors = 200697404)